src/lm.jl

```diff
@@ -206,17 +206,18 @@ function predict(mm::LinearModel, newx::AbstractMatrix;
     end
     if interval === nothing
         return retmean
-    elseif interval !== :confidence
-        error("only interval=:confidence is currently implemented") #:predint will be implemented
+    elseif !(interval ∈ (:confidence, :prediction))
+        error("only :confidence and :prediction intervals are defined")
     end
     length(mm.rr.wts) == 0 || error("prediction with confidence intervals not yet implemented for weighted regression")
 
     R = cholesky!(mm.pp).U #get the R matrix from the QR factorization
     residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))
     retvariance = (newx/R).^2 * residvar
+    interval == :prediction && (retvariance .+= deviance(mm)/dof_residual(mm))
```

Comment:
I'd find this more readable with an `if` block. You could also copy the full `(newx/R).^2 * residvar` part so that it's fused with the squaring step.

Reply:
I did try a few times to include it in the `(newx/R)` block, but found them less readable. I also had a go with the ternary operator
```
retvariance = interval == :prediction ? (newx/R).^2 * residvar .+ deviance(mm)/dof_residual(mm)) : (newx/R).^2 * residvar
```
Do you prefer that?

Reply:
The fusion doesn't matter as `deviance(mm)/dof_residual(mm)` is scalar?

Reply:
It would avoid a second pass over the data (if you also use `.*`, that is). To make it more readable, use an `if` block.

Reply:
no - that's a vector-matrix multiply, it shouldn't be broadcast.
I've pushed a commit that rewrites it to use an if block.